### PR TITLE
Fix for zero length cuepoints

### PR DESCRIPTION
**Pull request by vdeshpande**
This PR is for avoiding zero length cuepoint errors.
We are adding a small time to the endTime of a cuepoint, so that zero length cues(same startTime and endTime cues) do not throw out errors.

### CODE REVIEW

File: src/play-item.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ const playItem = (player, delay, item) => {
     let trackEl = player.addRemoteTextTrack({ kind: 'metadata' });
 
     item.cuePoints.forEach(cue => {
-      let vttCue = new Cue(cue.time, cue.time, cue.type);
+      let vttCue = new Cue(cue.startTime, cue.endTime, cue.type);
```
Comment (gkatsev) [inline src/play-item.js]: isn't this a breaking change to the format that theses cues should be?

**Comment by vdeshpande** [comment]
@gkatsev Yes, it's a breaking change in the current format of things, but it is a desired change in order to fix the issue.

File: src/play-item.js (hunk 1 above)
Comment (dmlap) [inline src/play-item.js]: You can make it more backwards compatible like this:

```js
let vttCue = new Cue(cut.time || cue.startTime || 0, cue.time || cue.endTime || 0, cue.type);
```

**Comment by dmlap** [comment]
I think it would be good to have one test that checks the legacy cue translation works correctly (i.e. `cue.time` instead of `startTime` and `endTime`).

**Comment by vdeshpande** [comment]
@dmlap I have added in the checks for compatibility.

File: test/play-item.test.js [hunk 2]
```diff
@@ -80,7 +81,70 @@ QUnit.test(
...
+    assert.equal(
+      setPoster,
+      'http://example.com/poster.png',
+      'poster is what we expected'
+    );
+
+    assert.deepEqual(
+      cues,
+      [{time: 0, type: 'foo' },
+      {time: 1, type: 'bar' }],
```
Comment (dmlap) [inline test/play-item.test.js]: I don't think this actually guarantees backwards compatibility. What I imagined you'd be testing is that cues like this `{ time: 14, type: 'whatever' }` would get translated into VTTCues like this `new VTTCue(14, 14.0001, 'whatever')`. It doesn't seem like that's occurring based on this assertion.

**Comment by vdeshpande** [comment]
@dmlap Made the change. Let me know if that change meets the criterion.

**Comment by vdeshpande** [comment]
@dmlap Let me know. Thanks!

**Comment by dmlap** [comment]
LGTM

**Pull request merged**